### PR TITLE
Again build on VS. Move CMake to extra.

### PR DESCRIPTION
**Pull request by acaly**
This time with tests. But there're 3 failed, and a stack corruption is reported.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
arc_unpacker_test.exe is a Catch v1.1 b3 host application.
Run with -? for options

-------------------------------------------------------------------------------
Decoding DXT5 DDS textures works
-------------------------------------------------------------------------------
D:\Documents\Programmes\cpp\arc_unpacker\tests\formats\microsoft\dds_converter_test.cc(28)
...............................................................................

D:\Documents\Programmes\cpp\arc_unpacker\tests\formats\microsoft\dds_converter_test.cc(28): FAILED:
due to unexpected exception with message:
  boost::filesystem::path codecvt to wstring: error

-------------------------------------------------------------------------------
Decoding DPNG images works
-------------------------------------------------------------------------------
D:\Documents\Programmes\cpp\arc_unpacker\tests\formats\qlie\dpng_converter_test.cc(6)
...............................................................................

D:\Documents\Programmes\cpp\arc_unpacker\tests\formats\qlie\dpng_converter_test.cc(6): FAILED:
due to unexpected exception with message:
  boost::filesystem::path codecvt to wstring: error

-------------------------------------------------------------------------------
Decoding TFCS files works
-------------------------------------------------------------------------------
D:\Documents\Programmes\cpp\arc_unpacker\tests\formats\touhou\tfcs_converter_test.cc(6)
...............................................................................

D:\Documents\Programmes\cpp\arc_unpacker\tests\test_support\converter_support.cc(30): FAILED:
  REQUIRE( expected_file.io.size() == actual_file.io.size() )
with expansion:
  3299 (0xce3) == 3285 (0xcd5)

===============================================================================
test cases:  113 |  110 passed | 3 failed
assertions: 1758 | 1755 passed | 3 failed

### CODE REVIEW

**Comment by acaly** [comment]
Related #18, #19.

**Comment by acaly** [comment]
Please check it. I'll add information in README later.

File: tests/formats/ivory/mbl_archive_test.cc [hunk 1]
```diff
@@ -8,7 +8,7 @@ static void test_mbl_archive(const std::string &path)
     std::shared_ptr<File> file1(new File);
     std::shared_ptr<File> file2(new File);
     file1->name = "abc.txt";
-    file2->name = "テスト";
```
Comment (rr-) [inline tests/formats/ivory/mbl_archive_test.cc]: nitpick: `code;//whatever` &rarr; `code; // whatever`

File: src/formats/microsoft/exe_archive.cc [hunk 2]
```diff
@@ -166,7 +166,7 @@ namespace
 }
 
 //keep flat hierarchy for unpacked files
-static const std::string path_sep = "／";
```
Comment (rr-) [inline src/formats/microsoft/exe_archive.cc]: nitpick: please add comment like in `mbl_archive_test`.

**Comment by rr-** [comment]
What is your file system encoding?

**Comment by acaly** [comment]
In fact I'm using [Boost Pre-Built Binaries](http://boost.teeks99.com/). I'm not sure what encoding it uses. If it's not related to VS support, I don't want to fix it now.

**Comment by rr-** [comment]
I mean, what is your system locale in Windows? Either way, I'll merge it after you add information to README.

After merging, I might also tweak it a little to support compiling with GNU g++ for people that, for whatever reason, don't want to use waf.

**Comment by acaly** [comment]
I'm using Chinese version of Windows 7. So it's GBK (936).

File: src/formats/microsoft/exe_archive.cc [hunk 3]
```diff
@@ -166,7 +166,7 @@ namespace
...
+static const std::string path_sep = "\xEF\xBC\x8F"; // "��"
```
Comment (rr-) [inline src/formats/microsoft/exe_archive.cc]: looks like the comment now has wrong encoding.

**Comment by rr-** [comment]
Can you check if `entrypoint.cc` resolves to 
https://github.com/vn-tools/arc_unpacker/blob/master/src/compat/entry_point.h#L11
rather than wmain in both tests and main executable?

**Comment by acaly** [comment]
Yes, _WIN32 is automatically defined, and it's resolved to wchat_t version.

**Comment by rr-** [comment]
Closed in https://github.com/vn-tools/arc_unpacker/commit/7dde78b15142cb2cb1db17a3ba3a78098bbf8425 .

**Pull request closed without merging**